oauthproxy.go

```diff
@@ -962,6 +976,7 @@ func (p *OAuthProxy) stripAuthHeaders(req *http.Request) {
 
 	if p.PassUserHeaders {
 		req.Header.Del("X-Forwarded-User")
+		req.Header.Del("X-Forwarded-Groups")
```

Comment:
It will need to be added just above in `p.PassBasicAuth` as well at the moment.  There's a bug that I want to fix in v7 where user headers are set when basic auth is set to true.

Can you add an entry or two to this test table making sure we strip these Groups headers appropriately in allowlisted paths? https://github.com/oauth2-proxy/oauth2-proxy/blob/2318716a8992686a31f4fadaa1ec3b12ff2dd22b/oauthproxy_test.go#L595

Reply:
Done (I hope)!